testing/osde2e/gpu-addon.sh

```diff
@@ -101,18 +98,18 @@ function addon_must_gather() {
     start=$SECONDS
     echo "Running the GPU Add-on must-gather"
 
-    addon_csv=$(oc get csv -n $ADDON_NAMESPACE -o custom-columns=NAME:.metadata.name --no-headers | grep nvidia-gpu-addon 2> /dev/null || true)
-    addon_must_gather_image=$(oc get csv "$addon_csv" -n $ADDON_NAMESPACE -o jsonpath='{.spec.relatedImages[?(@.name == "must-gather")].image}' --ignore-not-found)
-
-    tmp_dir="$(mktemp -d -t gpu-addon_XXXX)"
+    addon_csv=$(oc get csv -n $ADDON_NAMESPACE -o name | grep nvidia-gpu-addon || true)
+    addon_must_gather_image=$(oc get "$addon_csv" -n $ADDON_NAMESPACE -o jsonpath='{.spec.relatedImages[?(@.name == "must-gather")].image}' --ignore-not-found || true)
 
     if [[ ! "$addon_must_gather_image" ]]; then
         report_must_gather_junit 1 "Failed to find a GPU Add-on must-gather image" "$(($SECONDS - start))s"
```

Comment:
Won't this always be false? (the added `|| true`).
Maybe we can remove this `if` block.

Reply:
Without ` || true` and with `set -e` the line fails the script. With `|| true` it continues to the next statement, but `addon_must_gather_image` remains empty and makes the if-condition exit gracefully, producing a junit file.